test/gleam/string_test.gleam

```diff
@@ -364,3 +364,57 @@ pub fn to_option_test() {
   |> string.to_option
   |> should.equal(Some("ok"))
 }
+
+pub fn first_test() {
+  ""
+  |> string.first
+  |> should.equal(None)
+
+  "gleam"
+  |> string.first
+  |> should.equal(Some("g"))
+}
+
+pub fn last_test() {
+  ""
+  |> string.last
+  |> should.equal(None)
+
+  "gleam"
+  |> string.last
+  |> should.equal(Some("m"))
+
+  "gleam "
+  |> string.last
+  |> should.equal(Some(" "))
```

Comment:
Could we see a test for `"եոգլի"`, `"Ńaïve"` and other unicode strings such as emojis? Thanks